numpy/core/tests/test_multiarray.py

```diff
@@ -449,6 +449,10 @@ def test_array(self):
     def test_array_empty(self):
         assert_raises(TypeError, np.array)
 
+    def test_0d_array(self):
+        assert np.array(3) == 3
+        assert np.array(10).shape == ()
```

Comment:
Ah, my "BadSequence" test is probably not necessary, then, I guess.  This is not what I was going for ;):
```
np.ones(np.array(3)).shape == (3,)
```
is what I was going for.  The reason is that `list(np.array(3))` fails, I expect `PySequence_Check(np.array(3))` is True.

Reply:
Testing `np.array(3)` in `test_scalar_intp_converter` and `test_scalar_intp_from_sequence ` would be perfect (and more than sufficient).

Reply:
Tentative fix in 5d11ffc2614707fedd376ce668ec45c7cdb28880, what do you think?

My doubt is that testing `np.array(3)` is not suitable to be inserted inside `test_scalar_intp_converter ` or `test_scalar_intp_from_sequence` since only one path between the two possible paths (`PyArray_IntpFromSequence` or `PyArray_IntpConverter`) is followed, the other test is redundant (and wrongly located in some sense). Am I missing something?